sympy/physics/units/util.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+
+"""
+Several methods to simplify expressions involving unit objects.
+"""
+
+from __future__ import division
+
+from sympy.physics.units.quantities import Quantity
+from sympy import Add, Mul, Pow, Function, Rational
+from sympy.core.compatibility import reduce
+from sympy.physics.units.dimensions import Dimension
+
+
+def dim_simplify(expr):
+    """
+    NOTE: this function could be deprecated in the future.
+
+    Simplify expression by recursively evaluating the dimension arguments.
+
+    This function proceeds to a very rough dimensional analysis. It tries to
+    simplify expression with dimensions, and it deletes all what multiplies a
+    dimension without being a dimension. This is necessary to avoid strange
+    behavior when Add(L, L) be transformed into Mul(2, L).
+    """
+
+    if isinstance(expr, Dimension):
+        return expr
+
+    if isinstance(expr, Pow):
+        return dim_simplify(expr.base)**dim_simplify(expr.exp)
+    elif isinstance(expr, Function):
+        return dim_simplify(expr.args[0])
+    elif isinstance(expr, Add):
+        if (all(isinstance(arg, Dimension) for arg in expr.args) or
+            all(arg.is_dimensionless for arg in expr.args if isinstance(arg, Dimension))):
+            return reduce(lambda x, y: x.add(y), expr.args)
+        else:
+            raise ValueError("Dimensions cannot be added: %s" % expr)
+    elif isinstance(expr, Mul):
+        return Dimension(Mul(*[dim_simplify(i).name for i in expr.args if isinstance(i, Dimension)]))
+
+    raise ValueError("Cannot be simplifed: %s", expr)
+
+
+def convert_to(expr, quantity):
+    """
+    Convert `expr` to the same expression with all of its units and quantities
+    represented as factors of `quantity`, whenever the dimension is compatible.
+
+    Examples
+    ========
+
+    >>> from sympy.physics.units import speed_of_light, meter, gram, \
+        second, day, mile, newton, kilogram, inch, centimeter, atomic_mass_constant
+    >>> from sympy.physics.units.definitions import kilometer
+    >>> from sympy.physics.units import convert_to
+    >>> convert_to(mile, kilometer)
+    25146*kilometer/15625
+    >>> convert_to(mile, kilometer).n()
+    1.609344*kilometer
+    >>> convert_to(speed_of_light, meter/second)
+    299792458*meter/second
+    >>> convert_to(day, second)
+    86400*second
+    >>> 3*newton
+    3*newton
+    >>> convert_to(3*newton, kilogram*meter/second**2)
```

Comment:
Either the previous two lines could be deleted or this reference to `3*newton` could be changed to `_`, couldn't it?